src/components/ProteinStructureView.vue

```diff
@@ -62,39 +89,81 @@ export default {
       'value',
       'volume'
     ],
-    selectionRepresentations: []
   }),
 
   computed: {
-    pdbs: function() {
-      if (!this.uniprotData) {
-        return []
-      }
-      return $('entry dbReference[type="PDB"]', this.uniprotData).map((i, element) => {
-        const $element = $(element)
-        return {
-          id: $element.attr('id'),
-          method: $element.find('property[type="method"]').first().attr('value'),
-          resolution: $element.find('property[type="resolution"]').first().attr('value'),
-          chains: $element.find('property[type="chains"]').first().attr('value')
+    selectionDataWithSelectedColorBy: function() {
+        return _.map(this.selectionData, (x) => ({
+          start_residue_number: x.start_residue_number,
+          end_residue_number: x.end_residue_number,
+          color: _.get(x, this.colorBy, '#000')
+        }))
+    },
+    alphaFoldData: function() {
+        if (!this.uniprotData) {
+          return []
         }
-      }).get().filter((pdb) => pdb.id != null)
-    }
+        return $('entry dbReference[type="AlphaFoldDB"]', this.uniprotData).map((i, element) => {
+          const $element = $(element)
+          return {
+            id: $element.attr('id'),
+            method: $element.find('property[type="method"]').first().attr('value'),
+            resolution: $element.find('property[type="resolution"]').first().attr('value'),
+            chains: $element.find('property[type="chains"]').first().attr('value')
+          }
+        }).get().filter((x) => x.id != null)
+    },
   },
 
   mounted: function() {
     this.render()
   },
 
+  setup(props) {
+    const colorBy = ref('mean.color')
+
+    watch(() => props.rowSelected, (newValue) => {
+      if (_.isNumber(newValue)) {
+        colorBy.value = [newValue, 'color']
+      } else {
+        colorBy.value = 'mean.color'
+      }
+    })
+
+    return {
+      colorBy,
+    }
+  },
+
   watch: {
-    pdbs: {
+    colorBy: {
+      handler: function() {
+        this.viewerInstance.visual.select({data: this.selectionDataWithSelectedColorBy})
+      },
+    },
+    selectedResidueRanges: {
+      handler: function(newValue) {
+        const selectedRanges = newValue.map((x) => ({
+          start_residue_number: x.start,
+          end_residue_number: x.end,
+          color: null,
+          focus: true
+        }))
+        this.viewerInstance.visual.select({data:[...this.selectionDataWithSelectedColorBy, ...selectedRanges]})
+        this.viewerInstance.visual.highlight({
+            data: selectedRanges,
+        })
+      },
+      deep: true,
+    },
```

Comment:
Given that `this.viewerInstance` begins the lifecycle of this component as null, might it be worthwhile to guard its accession with `if (this.viewerInstance) { ... }`?